CacheManager.StackExchange.Redis.SelfCleaning.Tests/SelfCleaningRedisIntegrationTests.cs

```diff
@@ -0,0 +1,247 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using CacheManager.Core;
+using CacheManager.Core.Internal;
+using CacheManager.StackExchange.Redis.SelfCleaning.Core;
+using CacheManager.StackExchange.Redis.SelfCleaning.Timers;
+using Moq;
+using NUnit.Framework;
+using StackExchange.Redis;
+
+namespace CacheManager.StackExchange.Redis.SelfCleaning.Tests
+{
+    [TestFixture]
+    public class SelfCleaningRedisIntegrationTests
+    {
+        private const double TIME_TO_LIVE_MILLISECONDS_DIFFERENCE_THRESHOLD = 100;
+
+        private Mock<IServer> _serverMock;
+        private Mock<IDatabase> _databaseMock;
+        private Mock<IConnectionMultiplexer> _connectionMock;
+
+        private IDictionary<RedisKey, (RedisValue[] Values, DateTime InsertionTime)> _fauxDatabase;
+
+        private TimeSpan _timeToLive;
+        private ICacheManager<DummyModel> _cache;
+
+        private ICollection<(CacheItemRemovedEventArgs Args, DateTime Time)> _onRemoveByHandleInvocations;
+
+        [OneTimeSetUp]
+        public void Setup()
+        {
+            _serverMock = new Mock<IServer>();
+            _databaseMock = new Mock<IDatabase>();
+            _connectionMock = new Mock<IConnectionMultiplexer>();
+
+            _fauxDatabase = new ConcurrentDictionary<RedisKey, (RedisValue[], DateTime)>();
+
+            SetupServerMock();
+            SetupDatabaseMock();
+            SetupConnectionMock();
+            SetupCache();
+        }
+
+        #region Setup Methods
+
+        private void SetupServerMock()
+        {
+            _serverMock.SetupGet(server => server.IsConnected).Returns(true);
+            _serverMock.SetupGet(server => server.Features).Returns(new RedisFeatures(Version.Parse("3.0.504")));
+            _serverMock
+                .Setup(server => server.Keys(It.IsAny<int>(), It.IsAny<RedisValue>(), It.IsAny<int>(), It.IsAny<long>(),
+                    It.IsAny<int>(), It.IsAny<CommandFlags>()))
+                .Returns<int, RedisValue, int, long, int, CommandFlags>(
```

Comment:
As we spoke, check if you can covert to `.Returns(_fauxDatabase.Keys);`.

If not, we might want a comment here? Or it will be too confusing and unnecessary?

Reply:
As I expected, that keeps the value `_fauxDatabase.Keys` had in that exact moments and disrupts the tests, BUT I was able to use an empty lambda (`() => ...`) instead.